best_genome_reference.snakefile

```diff
@@ -0,0 +1,286 @@
+    shell:'''
+    sourmash sketch dna -p k=21,k=31,scaled=1000,abund -o {output} --name {wildcards.sra} {input}
+    ''' 
+
+rule sourmash_download_gtdb_database_k31:
+    output: "inputs/sourmash_dbs/gtdb-rs202.genomic.k31.zip" 
+    resources:
+        mem_mb = 1000,
+        tmpdir = TMPDIR
+    threads: 1
+    shell:'''
+    wget -O {output} https://osf.io/94mzh/download
+    '''
+
+rule sourmash_download_gtdb_database_k21:
+    output: "inputs/sourmash_dbs/gtdb-rs202.genomic.k21.zip" 
+    resources:
+        mem_mb = 1000,
+        tmpdir = TMPDIR
+    threads: 1
+    shell:'''
+    wget -O {output} https://osf.io/vgex4/download
+    '''
+
+rule sourmash_download_human_sig:
+    output: "inputs/sourmash_dbs/GCF_000001405.39_GRCh38.p13_rna.sig"
+    resources:
+        mem_mb = 1000,
+        tmpdir = TMPDIR
+    threads: 1
+    shell:'''
+    wget -O {output} https://osf.io/anj6b/download
+    '''
+
+rule rnaseq_sample_sourmash_gather_against_gtdb:
+    input: 
+        sig="outputs/rnaseq_sourmash_sketch/{sra}.sig",
+        db="inputs/sourmash_dbs/gtdb-rs202.genomic.k{ksize}.zip",
+        human="inputs/sourmash_dbs/GCF_000001405.39_GRCh38.p13_rna.sig"
+    output: "outputs/rnaseq_sourmash_gather/{sra}_gtdb_k{ksize}.csv"
+    resources:
+        mem_mb = lambda wildcards, attempt: attempt * 16000 ,
+        tmpdir= TMPDIR
+    threads: 1
+    benchmark: "benchmarks/rnaseq/sourmash_gather_k{ksize}_{sra}.txt"
+    conda: "envs/sourmash.yml"
+    shell:'''
+    sourmash gather -o {output} --scaled 2000 -k {wildcards.ksize} {input.sig} {input.db} {input.human}
+    '''
+
+rule rnaseq_sample_sourmash_search_against_gtdb:
+    input: 
+        sig="outputs/rnaseq_sourmash_sketch/{sra}.sig",
+        db="inputs/sourmash_dbs/gtdb-rs202.genomic.k{ksize}.zip",
+    output: "outputs/rnaseq_sourmash_search/{sra}_gtdb_k{ksize}_max_containment.csv"
+    resources:
+        mem_mb = lambda wildcards, attempt: attempt * 16000 ,
+        tmpdir= TMPDIR
+    threads: 1
+    benchmark: "benchmarks/rnaseq/sourmash_search_k{ksize}_{sra}.txt"
+    conda: "envs/sourmash.yml"
+    shell:'''
+    sourmash search --max-containment -o {output} --scaled 2000 -k {wildcards.ksize} {input.sig} {input.db} 
+    '''
+
+checkpoint rnaseq_sample_select_best_genome_reference:
+    """
+    I think this will read in the gather results, and output 
+    one dataframe per RNA seq sample. The dataframe will have the SRA
+    accession and the reference genome accession. wildcard {sra} will already
+    be defined as a list of RNAseq SRA accessions at the beginning of the 
+    Snakefile, and the checkpoint class will output an amalgamation of wildcards
+    {sra}-{acc}, which will be called {sra_to_acc} in the rule all, but the hyphen
+    should allow for the wildcards to be solved separately. 
+    Example:
+    https://github.com/taylorreiter/2021-metapangenome-example/blob/main/Snakefile#L90
+    """
+
+rule reference_make_accession_info_csv:
```

Comment:
What does a checkpoint without input or output do?

Reply:
absolutely nothing 😆 this is a vestige of the direction this snakefile was going, before I saw the intermediate gather outputs and sort of conceptually realized this pipeline wasn't going to work. I had thought that there was a chance that our reference collection was good enough to have a reference genome for almost everything that we had isolate RNAseq for, but even on the ~13 libraries I tested, this was only true for half of them (which was _not_ what I expected). So the checkpoint is a placeholder for what I would do in the future, but haven't done it yet. I'll either make a note of this somewhere or comment it out. Thank you!